src/RoomsSDKAdapter.test.js

```diff
@@ -1,83 +1,186 @@
       const errorMessage = 'a proper error message';
 
       mockSDK.rooms.get = jest.fn(() => Promise.reject(new Error(errorMessage)));
 
-      roomsSDKAdapter.getRoom('id').subscribe(
-        () => {},
-        (error) => {
-          expect(error.message).toBe(errorMessage);
-          done();
-        },
-      );
+      roomsSDKAdapter.getRoom('id')
+        .subscribe(
+          () => {
+          },
+          (error) => {
+            expect(error.message)
+              .toBe(errorMessage);
+            done();
+          },
+        );
     });
   });
 
   describe('getRoomActivities()', () => {
     test('returns an observable', () => {
-      expect(isObservable(roomsSDKAdapter.getRoomActivities())).toBeTruthy();
+      expect(isObservable(roomsSDKAdapter.getRoomActivities()))
+        .toBeTruthy();
     });
 
     test('returns a activity in a proper shape', (done) => {
       mockSDK.internal.mercury.on = jest.fn((event, callback) => callback(mockSDKActivity));
 
-      roomsSDKAdapter.getRoomActivities(mockSDKActivity.target.id).subscribe((activity) => {
-        expect(activity).toEqual({
-          ID: mockSDKActivity.id,
-          roomID: mockSDKActivity.target.id,
-          content: mockSDKActivity.object,
-          contentType: mockSDKActivity.object.objectType,
-          personID: mockSDKActivity.actor.id,
-          displayAuthor: false,
-          created: mockSDKActivity.published,
+      roomsSDKAdapter.getRoomActivities(mockSDKActivity.target.id)
+        .subscribe((activity) => {
+          expect(activity)
+            .toEqual({
+              ID: mockSDKActivity.id,
+              roomID: mockSDKActivity.target.id,
+              content: mockSDKActivity.object,
+              contentType: mockSDKActivity.object.objectType,
+              personID: mockSDKActivity.actor.id,
+              displayAuthor: false,
+              created: mockSDKActivity.published,
+            });
+          done();
+        });
+    });
+
+    test('throws error if no room id is present', (done) => {
+      roomsSDKAdapter.getPreviousActivities()
+        .subscribe({
+          next() {
+          },
+          error(e) {
+            console.log(e);
+            expect(e)
+              .toEqual(new Error('getPreviousActivities - Must provide room ID'));
+            done();
+          },
+        });
+    });
+
+    test('sets empty roomActivities if no room exists', () => {
+      expect(roomsSDKAdapter.roomActivities.has('room-1'))
+        .toBe(false);
+      roomsSDKAdapter.getPreviousActivities('room-1');
+      expect(roomsSDKAdapter.roomActivities.has('room-1'))
+        .toBe(true);
+      expect(roomsSDKAdapter.roomActivities.get('room-1'))
+        .toStrictEqual({
+          activities: new Map(),
+          earliestActivityDate: null,
+          activityIds: new Map(),
         });
-        done();
+    });
+  });
+
+  describe('getPreviousActivities() functionality', () => {
```

Comment:
Let's keep titles consistent. I do see that `getRooms` above also has "functionality" in it, but I think this is a one-off that slipped through, the rest of the code base doesn't add that
```suggestion
  describe('getPreviousActivities()', () => {
```